website/docs/test_core_lib.md

```diff
@@ -0,0 +1,187 @@
+
+`user_data_access.py`
+```python
+from http import HTTPStatus
+
+from core_lib.data_layers.data_access.db.crud.crud_data_access import CRUDDataAccess
+from core_lib.connection.sql_alchemy_connection_registry import SqlAlchemyConnectionRegistry
+from core_lib.error_handling.status_code_exception import StatusCodeException
+from user_core_lib.data_layers.data.db.user import User
+from core_lib.error_handling.not_found_decorator import NotFoundErrorHandler
+
+
+class UserDataAccess(CRUDDataAccess):
+    def __init__(self, db: SqlAlchemyConnectionRegistry):
+        CRUD.__init__(self, User, db)
+```
+
+## Service 
+Here we right the business logic for the extracted data, as the `DataAccess` only handles the interfacing with the connection, a `Service` will only be used to take the data and process it.
+
+`user_service.py`
+```python
+from core_lib.data_transform.result_to_dict import ResultToDict
+from core_lib.data_layers.service.service import Service
+from user_core_lib.data_layers.data_access.user_data_access import UserDataAccess
+
+
+class UserService(Service):
+    def __init__(self, data_access: UserDataAccess):
+        self.data_access = data_access
+
+    @ResultToDict()
+    def create(self, user_data: dict):
+        return self.data_access.create(user_data)
+
+    @ResultToDict()
+    def get(self, user_id: int):
+        return self.data_access.get(user_id)
+
+    def update(self, user_id: int, update: dict):
+        return self.data_access.update(user_id, update)
+
+    def delete(self, user_id: int):
+        return self.data_access.delete(user_id)
+```
+
+## Initializing
+For initializing our `Core-Lib` we will need a config that is stored under the config directory.
+
+`user_core_lib.yaml`
+```yaml
+# @package _global_
+core_lib:
+  user_core_lib:
+    cache:
+      memory_cache:
+        type: memory
+    data:
+      userdb:
+        log_queries: false
+        create_db: true
+        session:
+          pool_recycle: 3200
+          pool_pre_ping: false
+        url:
+          file: ${oc.env:USERDB_DB}
+          protocol: postgresql
+          username: ${oc.env:USERDB_USER}
+          password: ${oc.env:USERDB_PASSWORD}
+          port: ${oc.env:USERDB_PORT}
+          host: ${oc.env:USERDB_HOST}
+    client:
+      user_client:
+        _target_: core_lib.client.client_base.ClientBase
+        base_url: https://example.com/
+```
+
+In your test file
+
+`test.py`
+```python
+import unittest
+
+from core_lib.error_handling.status_code_exception import StatusCodeException
+from user_core_lib.user_core_lib import UserCoreLib
+from tests.test_data.test_utils import sync_create_core_lib_config
+from core_lib.client.client_base import ClientBase
+from core_lib.helpers.config_instances import instantiate_config
+
+class UserClient(ClientBase):
```

Comment:
1. on UserClient
get will use the self._get(f'/user/{user_id}')

create `UserClientMock`? 
define this class for the config file for testing